File: kide-enterprise/backend/app/schemas/ai.py
from typing import List, Optional, Dict, Any
from pydantic import BaseModel, ConfigDict

class ChatMessage(BaseModel):
    role: str # user, assistant, system
    content: str

class ToolCallRecord(BaseModel):
    id: str
    tool: str
    input: Dict[str, Any]
    output: Any
    status: str = "success"

class ProposedPatch(BaseModel):
    filename: str
    action: str = "modify" # modify, create, delete
    diff: str
    new_content: str
    rationale: str

class AIChatRequest(BaseModel):
    project_id: int
    messages: List[ChatMessage]
    context_file: Optional[str] = None
    provider: Optional[str] = None
    model: Optional[str] = None

class AIChatResponse(BaseModel):
    session_id: str
    provider: str
    model: str
    message: str
    tool_calls: List[ToolCallRecord] = []
    proposed_patches: List[ProposedPatch] = []
    provenance_id: Optional[int] = None

class ApplyPatchRequest(BaseModel):
    project_id: int
    provenance_id: Optional[int] = None
    patches: List[ProposedPatch]

class ApplyPatchResponse(BaseModel):
    status: str
    applied_files: List[str]
    timestamp: str

class AIProvenanceResponse(BaseModel):
    id: int
    project_id: int
    session_id: str
    provider: str
    model_name: str
    user_prompt: str
    tool_calls: Optional[List[Dict[str, Any]]] = None
    assistant_response: str
    proposed_patches: Optional[List[Dict[str, Any]]] = None
    status: str
    created_at: str

    model_config = ConfigDict(from_attributes=True)

class AIProvidersResponse(BaseModel):
    available_providers: List[str]
    active_provider: str
    active_model: str
    has_gemini_key: bool
    has_openai_key: bool
    has_anthropic_key: bool

File: kide-enterprise/frontend/src/api/ai.ts
import { fetchClient } from './client';

export interface ChatMessage {
  id?: string;
  role: 'user' | 'assistant' | 'system';
  content: string;
}

export interface ToolCallRecord {
  id: string;
  tool: string;
  input: Record<string, any>;
  output: any;
  status: string;
}

export interface ProposedPatch {
  filename: string;
  action: 'modify' | 'create' | 'delete';
  diff: string;
  new_content: string;
  rationale: string;
}

export interface AIChatResponse {
  session_id: string;
  provider: string;
  model: string;
  message: string;
  tool_calls: ToolCallRecord[];
  proposed_patches: ProposedPatch[];
  provenance_id?: number;
}

export interface ApplyPatchResponse {
  status: string;
  applied_files: string[];
  timestamp: string;
}

export interface AIProvenanceRecord {
  id: number;
  project_id: number;
  session_id: string;
  provider: string;
  model_name: string;
  user_prompt: string;
  tool_calls?: any[];
  assistant_response: string;
  proposed_patches?: any[];
  status: string;
  created_at: string;
}

export interface AIProvidersResponse {
  available_providers: string[];
  active_provider: string;
  active_model: string;
  has_gemini_key: boolean;
  has_openai_key: boolean;
  has_anthropic_key: boolean;
}

export async function sendAiMessage(
  projectId: number,
  messages: ChatMessage[],
  contextFile?: string,
  provider?: string,
  model?: string
): Promise<AIChatResponse> {
  return fetchClient('/ai/chat', {
    method: 'POST',
    body: JSON.stringify({
      project_id: projectId,
      messages,
      context_file: contextFile,
      provider,
      model,
    }),
  });
}

export async function applyAiPatch(
  projectId: number,
  provenanceId: number | undefined,
  patches: ProposedPatch[]
): Promise<ApplyPatchResponse> {
  return fetchClient('/ai/apply-patch', {
    method: 'POST',
    body: JSON.stringify({
      project_id: projectId,
      provenance_id: provenanceId,
      patches,
    }),
  });
}

export async function getAiProvenance(projectId: number): Promise<AIProvenanceRecord[]> {
  return fetchClient(`/ai/provenance/${projectId}`);
}

export async function getAiProviders(): Promise<AIProvidersResponse> {
  return fetchClient('/ai/providers');
}
